### PR TITLE
docs: add necessary steps to angular and vue authenticator guides

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes

Some docs updates for Vue and Angular Authenticator to make it more clear that there are additional necessary setup steps (currently only called out in troubleshooting). This adds the necessary polyfill/config updates to the quick start steps on the Authenticator pages. Preview links: [Vue](https://addpolyfillsteps.dvmvffzts1tcu.amplifyapp.com/vue/connected-components/authenticator) | [Angular](https://addpolyfillsteps.dvmvffzts1tcu.amplifyapp.com/angular/connected-components/authenticator) Other Auth quick start guides should be unchanged. 

Fixes https://github.com/aws-amplify/amplify-ui/issues/3946
Fixes https://github.com/aws-amplify/amplify-ui/issues/3501

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/connected-components/authenticator/quick-start.vue.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+### Step 3. Update Vite config and index.html
+
+When working with Vite in a Vue project, you must make a few modifications. Please follow the steps below.
+
+<Fragment>
+  {({ platform }) => import('../../getting-started/troubleshooting/shared/vite-polyfill.mdx')}
+</Fragment>
+
+<Fragment>
+  {({ platform }) => import('../../getting-started/troubleshooting/shared/vite-polyfill-ts.mdx')}
```
Comment (reesscot) [inline docs/src/pages/[platform]/connected-components/authenticator/quick-start.vue.mdx:24]: I think this piece needs to show conditionally inside of a TypeScript Tab?

Comment (hbuchel) [reply docs/src/pages/[platform]/connected-components/authenticator/quick-start.vue.mdx:24]: Oh or do you mean it should be like on the React troubleshooting steps with Vite? https://ui.docs.amplify.aws/react/getting-started/troubleshooting#vite

Comment (hbuchel) [reply docs/src/pages/[platform]/connected-components/authenticator/quick-start.vue.mdx]: This is fixed in latest commits and updated here: https://addpolyfillsteps.dvmvffzts1tcu.amplifyapp.com/vue/connected-components/authenticator#step-3-update-vite-config-and-indexhtml

File: docs/src/pages/[platform]/getting-started/troubleshooting/shared/angular-polyfill.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  "files": [
+    "src/main.ts",
+    "src/polyfills.ts"
+  ],
+}
+```
+
+  </TabItem>
+
+  <TabItem title="Angular 11-14">
```
Comment (sreeramsama) [inline docs/src/pages/[platform]/getting-started/troubleshooting/shared/angular-polyfill.mdx:39]:
```suggestion
  <TabItem title="Angular 12-14">
```

Comment (sreeramsama) [reply docs/src/pages/[platform]/getting-started/troubleshooting/shared/angular-polyfill.mdx:39]: we are on Angular 12 on `main`

Comment (hbuchel) [reply docs/src/pages/[platform]/getting-started/troubleshooting/shared/angular-polyfill.mdx]: @sreeramsama thanks, updated

**Review by sreeramsama — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM, smoke tested https://addpolyfillsteps.dvmvffzts1tcu.amplifyapp.com/flutter/connected-components/authenticator

**Pull request merged**